### DISCUSSION TITLE
[Feature Request] NuGet Symbol Server Support

### DISCUSSION THREAD

**Discussion by grochoge** [community/community#69837] [Packages]
### Select Topic Area

Product Feedback

### Body

Please add support for acting a symbol server to serve Windows and "Portable" PDB files from .snupkg and .symbols.nupkg archives, as is supported by [Artifactory](https://jfrog.com/help/r/jfrog-artifactory-documentation/artifactory-as-your-symbol-server).

Note that .symbols.nupkg and Windows PDBs are described as "legacy" in some Microsoft documentation, but they're the only option for C++/native symbols.

**Comment by esurov** [comment]
Dear @grochoge ,

I don't have the ability to directly add support for specific features or functionality in Github. However, I can provide you with guidance on how you might go about implementing support for acting as a symbol server for Windows and "Portable" PDB files from .snupkg and .symbols.nupkg archives.

To create a symbol server that serves Windows and "Portable" PDB files from .snupkg and .symbols.nupkg archives, you'll need to consider a few key steps:

1. **Server Setup:**
   You'll need to set up a server to host the symbol files. This can be a web server or a specialized symbol server software. Many organizations use IIS (Internet Information Services) for hosting symbol files on Windows servers.

2. **Symbol Packaging:**
   Ensure that your .snupkg and .symbols.nupkg archives contain the necessary PDB and related symbol files. The structure of these packages should follow the guidelines provided by the .NET platform.

3. **Symbol Server Configuration:**
   Configure your server to understand requests for symbol files and their corresponding PDB files. You might need to modify the server settings to recognize specific file extensions and directory structures, and route requests accordingly.

4. **Symbol Server Path Structure:**
   Organize your symbol files in a directory structure that corresponds to the directory structure found in the .snupkg and .symbols.nupkg archives. This is important for matching requests to the correct symbol files.

5. **Handling Symbol Requests:**
   When a symbol file request is received, your server should locate the corresponding symbol file within the .snupkg or .symbols.nupkg archive, extract it, and serve it to the client. This may require custom server-side scripting or code.

6. **Client Configuration:**
   Clients, such as debuggers, need to be configured to use your symbol server. You'll need to specify the symbol server's URL and any authentication if required.

7. **Testing:**
   Thoroughly test the symbol server to ensure it correctly serves symbol files to clients. Debug a sample application using your symbol server to verify that symbols are being loaded correctly.

8. **Security and Access Control:**
   Implement access controls and security measures to ensure that only authorized users and applications can access the symbol server.

9. **Monitoring and Maintenance:**
   Set up monitoring and maintenance processes to ensure the symbol server remains operational and responsive. This includes regularly updating the symbol files in the .snupkg and .symbols.nupkg archives as new versions of software are released.

Remember that support for Windows PDB files and .symbols.nupkg archives may require specific knowledge of the .NET platform, as well as the ability to handle these file formats effectively. You may also want to refer to the latest Microsoft documentation and guidelines for symbol server setup and configuration.
